flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/EdgeManagerBuildUtil.java

```diff
@@ -171,4 +177,30 @@ private static void connectPointwise(
             }
         }
     }
+
+    private static ConsumedPartitionGroup createAndRegisterConsumedPartitionGroupToEdgeManager(
+            IntermediateResultPartitionID consumedPartitionId,
+            IntermediateResult intermediateResult) {
+        ConsumedPartitionGroup consumedPartitionGroup =
+                ConsumedPartitionGroup.fromSinglePartition(consumedPartitionId);
+        intermediateResult
+                .getProducer()
+                .getGraph()
+                .getEdgeManager()
+                .registerConsumedPartitionGroup(consumedPartitionGroup);
+        return consumedPartitionGroup;
```

Comment:
The method can be implemented as `createAndRegisterConsumedPartitionGroupToEdgeManager(Collections.singleton(consumedPartitionId))`